vms/secp256k1fx/keychain.go

```diff
@@ -143,8 +166,12 @@ func (kc *Keychain) String() string {
 
 // to avoid internals type assertions
 func (kc Keychain) get(id ids.ShortID) (*secp256k1.PrivateKey, bool) {
-	if i, ok := kc.addrToKeyIndex[id]; ok {
+	if i, ok := kc.avaxAddrToKeyIndex[id]; ok {
 		return kc.Keys[i], true
 	}
 	return nil, false
 }
+
+func publicKeyToEthAddress(pk *secp256k1.PublicKey) common.Address {
+	return crypto.PubkeyToAddress(*(pk.ToECDSA()))
+}
```

Comment:
This is also implemented in coreth - but would cause a circular import. I'd be willing to expose this here if we wanted coreth to just depend on this and we remove the duplicated code